Marlin/stepper.cpp

```diff
@@ -492,22 +472,31 @@ ISR(TIMER1_COMPA_vect) {
     // Check endstops
     if (check_endstops) {
       
-      #define _ENDSTOP(axis, minmax) axis ##_## minmax ##_endstop
+      char current_endstop_bits;
+      #ifdef Z_DUAL_ENDSTOPS
+        char current_dual_endstop_bits;
```

Comment:
I think we should just add a Z2 bit to the `current_endstop_bits`. That's a total of 7 endstop bits, so still fits.